fluentinterface/src/main/java/com/iluwatar/fluentinterface/fluentiterable/simple/SimpleFluentIterable.java

```diff
@@ -0,0 +1,194 @@
+package com.iluwatar.fluentinterface.fluentiterable.simple;
+
+import com.iluwatar.fluentinterface.fluentiterable.FluentIterable;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * This is a simple implementation of the FluentIterable interface. It evaluates
+ * all chained operations eagerly.
+ * @param <TYPE> the type of the objects the iteration is about
+ */
+public class SimpleFluentIterable<TYPE> implements FluentIterable<TYPE> {
+
+    private final Iterable<TYPE> iterable;
+
+    /**
+     * This constructor creates a copy of a given iterable's contents.
+     * @param iterable the iterable this interface copies to work on.
+     */
+    protected SimpleFluentIterable(Iterable<TYPE> iterable) {
```

Comment:
* The actual reason is homogenity: For some implementations of the FluentIterable, more than one constructor is required (for example LazyFluentIterable). In order to prevent calling constructors of FluentIterable implementations directly, they are provided via static factory methods on the implementing class.
In this case, a public constructor could be provided, but then again, it should be provided for the lazy implementation as well - and for the lazy class, there are currently two constructors of which at least one has to be protected (the parameterless one). Would it be better to switch to public constructors?
* It would be possible to use an incoming _iterator_ - but it would be problematic if an _iterable_ is provided, because this would be misleading for the user of our implementation. If an iterator was provided, the user could expect that no defensive copy was made, but it would have made some of the example methods more complicated, I think. Do you agree?